backend/lib/src/api/handlers.rs

```diff
@@ -274,6 +278,7 @@ pub async fn get_file_info(
 /// and returns a mock success response without actual file processing.
 ///
 /// TODO: Further optimize this to avoid having to load the entire file into memory.
+#[cfg_attr(feature = "mocks", allow(unused_variables))]
```

Comment:
This is the same comment I made to Herman - even tho with mock mode we enable mock rpc and mock db we should instead add this mocking at that level and leave the handler code non-mock. 
Otherwise we end up testing not the real code path when in mock mode